tests/bgp/test_bgp_session_flap.py

```diff
@@ -0,0 +1,211 @@
+'''This script is to test BGP session flapping on SONiC and monitor
+the CPU.
+'''
+import logging
+
+import pytest
+import time
+from tests.common.utilities import InterruptableThread
+import textfsm
+import traceback
+
+from natsort import natsorted
+
+logger = logging.getLogger(__name__)
+max_wait = 0
+wait_time = 2
+stop_threads = False
+proc_textfsm = "./bgp/templates/show_proc_cpu.template"
+bgp_sum_textfsm = "./bgp/templates/bgp_summary.template"
+flap_threads = []
+skip_hosts = []
+
+cpuSpike = 10
+memSpike = 1.3
+
+pytestmark = [
+    pytest.mark.topology('t1')
+]
+
+
+def get_cpu_stats(dut):
+    proc_cpu = dut.shell("show processes cpu | head -n 20", module_ignore_errors=True)['stdout']
+    proc_mem = dut.shell("show processes memory | head -n 20", module_ignore_errors=True)['stdout']
+    proc_sum = dut.shell("show processes summary | grep -v '0.0 0.0'", module_ignore_errors=True)['stdout']
+    bgp_cpu = dut.shell("show processes cpu | grep bgp", module_ignore_errors=True)['stdout']
+    bgp_v4_sum = dut.shell("show ip bgp summary | grep memory", module_ignore_errors=True)['stdout']
+    bgp_v6_sum = dut.shell("show ipv6 bgp summary | grep memory", module_ignore_errors=True)['stdout']
+    logger.info("CPU:\n{}\nMemory:\n{}\nSummary:\n{}\nBGP Memory:\n{}\nBGP IPv4:\n{}\nIPv6:\n{}\n"
+                .format(proc_cpu, proc_mem, proc_sum, bgp_cpu, bgp_v4_sum, bgp_v6_sum))
+    with open(proc_textfsm) as template:
+        fsm = textfsm.TextFSM(template)
+        parsed_cpu = fsm.ParseText(proc_cpu)[0]
+
+    with open(bgp_sum_textfsm) as template:
+        fsm = textfsm.TextFSM(template)
+        parsed_ipv4 = fsm.ParseText(bgp_v4_sum)[0]
+        parsed_ipv6 = fsm.ParseText(bgp_v6_sum)[0]
+    data = [float(parsed_cpu[0]), float(parsed_cpu[1]), float(parsed_cpu[2]),
+            float(parsed_ipv4[0]), float(parsed_ipv4[1]), float(parsed_ipv4[2]),
+            float(parsed_ipv6[0]), float(parsed_ipv6[1]), float(parsed_ipv6[2])]
+    return data
+
+
+@pytest.fixture(scope='module')
+def setup(tbinfo, nbrhosts, duthosts, rand_one_dut_hostname, enum_rand_one_frontend_asic_index):
+    duthost = duthosts[rand_one_dut_hostname]
+    namespace = duthost.get_namespace_from_asic_id(enum_rand_one_frontend_asic_index)
+
+    tor_neighbors = dict()
+    # tor1 = natsorted([neighbor for neighbor in nbrhosts.keys() if neighbor.endswith('T0')])[0]
+    tor1 = natsorted(nbrhosts.keys())[0]
+
+    bgp_facts = duthost.bgp_facts(instance_id=enum_rand_one_frontend_asic_index)['ansible_facts']
+    neigh_asn = dict()
+    for k, v in bgp_facts['bgp_neighbors'].items():
+        if v['description'] not in skip_hosts:
+            neigh_asn[v['description']] = v['remote AS']
+            tor_neighbors[v['description']] = nbrhosts[v['description']]["host"]
```

Comment:
Hi.. The test is failing on Multi-Asic T1 DUT. Reason is that on multiAsic DUTs there are iBGP sessions b/w ASICs. These are not present in nbrhosts. and hence setup fails at Line#68.

You might want to skip iBGP neighbors(populate skip hosts)?? May be use duthost.get_asic_namespace_list() from tests/common/devices/multi_asic.py to populate skip_hosts..

Reply:
Thanks @Azarack .. Have you verified the changes on multi-asic DUT?

Reply:
I have not been able to verify with a multi-asic DUT as I do not have one available to me currently.

Reply:
@Azarack , it still fails for multi-DUT..

![image](https://user-images.githubusercontent.com/115033986/227365776-2a105fce-a322-46f1-a933-08b475beaa93.png)

>> Internal Neighbors in bgp-facts are in CAPS, where as in skip_hosts its populated in non-Caps. 
May be update this?
>>>> if v['description'] not in skip_hosts:
   to if v['description'].lower() not in skip_hosts??

Reply:
I have made this change.